main-actions/src/main/scala/sbt/Doc.scala

```diff
@@ -23,93 +19,45 @@ import sbt.internal.util.ManagedLogger
 
 object Doc {
   import RawCompileLike._
+
   def scaladoc(label: String,
                cacheStoreFactory: CacheStoreFactory,
                compiler: AnalyzingCompiler): Gen =
     scaladoc(label, cacheStoreFactory, compiler, Seq())
+
   def scaladoc(label: String,
                cacheStoreFactory: CacheStoreFactory,
                compiler: AnalyzingCompiler,
                fileInputOptions: Seq[String]): Gen =
     cached(cacheStoreFactory,
            fileInputOptions,
            prepare(label + " Scala API documentation", compiler.doc))
-  def javadoc(label: String,
-              cacheStoreFactory: CacheStoreFactory,
-              doc: JavaTools,
-              log: Logger,
-              reporter: Reporter): Gen =
-    javadoc(label, cacheStoreFactory, doc, log, reporter, Seq())
-  def javadoc(label: String,
-              cacheStoreFactory: CacheStoreFactory,
-              doc: JavaTools,
-              log: Logger,
-              reporter: Reporter,
-              fileInputOptions: Seq[String]): Gen =
-    cached(
-      cacheStoreFactory,
-      fileInputOptions,
-      prepare(
-        label + " Java API documentation",
-        filterSources(
-          javaSourcesOnly,
-          (sources: Seq[File],
-           classpath: Seq[File],
-           outputDirectory: File,
-           options: Seq[String],
-           maxErrors: Int,
-           log: Logger) => {
-            // doc.doc
-            ???
-          }
-        )
-      )
-    )
 
+  @deprecated("Going away", "1.1.1")
+  def javadoc(
+      label: String,
+      cacheStoreFactory: CacheStoreFactory,
+      doc: JavaTools,
+      log: Logger,
+      reporter: Reporter,
+  ): Gen = ???
```

Comment:
Why `???` instead of removing it or delegating it to somewhere?

Reply:
It's public API, and I think a non-fatal runtime error, that can be handled, is better than a fatal linking error.

Reply:
I don't think we should break this until sbt 2.x unless there's a really good reason for it.

Reply:
It's already broken, I just made it more obviously so (and got rid of a bunch of unused warnings in the process):

https://github.com/sbt/sbt/blob/v1.1.0-RC1/main-actions/src/main/scala/sbt/Doc.scala#L63